src/main/java/graphql/annotations/processor/util/DataFetcherConstructor.java

```diff
@@ -38,14 +38,31 @@ public DataFetcher constructDataFetcher(String fieldName, GraphQLDataFetcher ann
         if (args.length == 0) {
             return newInstance(annotatedDataFetcher.value());
         } else {
-            try {
-                final Constructor<? extends DataFetcher> ctr = annotatedDataFetcher.value().getDeclaredConstructor(
-                        stream(args).map(v -> String.class).toArray(Class[]::new));
-                return constructNewInstance(ctr, (Object[]) args);
-            } catch (final NoSuchMethodException e) {
-                throw new GraphQLAnnotationsException("Unable to instantiate DataFetcher via constructor for: " + fieldName, e);
-            }
+            return Stream.of(annotatedDataFetcher.value().getConstructors())
+                    // filter only constructor that have the same args as the annotation args
+                    // or that get String[], String... args
+                    .filter(x -> (x.getParameterCount() == args.length) ||
+                            (x.getParameterTypes().length == 1 && x.getParameterTypes()[0] == String[].class))
+                    .map(x -> {
+                        try {
+                            Constructor<? extends DataFetcher> constructor;
+                            if (x.getParameterTypes().length == 1 && x.getParameterTypes()[0] == String[].class) {
+                                constructor = annotatedDataFetcher.value().getDeclaredConstructor(String[].class);
+                                return constructNewInstance(constructor, new Object[]{args});
```

Comment:
new Object[]{args} -> (Object[]) args

Reply:
it's in purpose that way 
new Object[]{args} create a single variable of object[],
and (Object[]) is a casting that casts to a object array, and the String[] ctor expecting only one argument...